pypreprocess/nipype_preproc_fsl_utils.py

```diff
@@ -180,3 +181,174 @@ def do_subject_preproc(subject_id,
         output['func'] = applyxfm_results.outputs.out_file
 
     return output
+
+
+def _do_img_topup_correction(ap_realig_img, pa_realig_img, func_imgs,
+                             total_readout_times=(1., 1.), memory=None,
+                             tmp_dir=None):
+    """
+    Compute and apply topup as implemented in FSL.
+
+    It is crucial to provide the total_readout_times of the ap_img and pa_img
+    correctly in case they are different. Otherwise a value of 1 can be taken
+    as default.
+
+    More detailed documentation can be found in the example provided in the FSL
+    webpage. http://fsl.fmrib.ox.ac.uk/fsl/fslwiki/TOPUP/ExampleTopupFollowedByApplytopup.
+
+    Parameters
+    ----------
+    ap_realig_img: string
+        path to img using negative phase-encode blips. Anterior --> Posterior
+        the image should be already realigned
+
+    pa_realig_img: string
+        path to img using positive phase-encode blips. Posterior --> Anterior
+        the image should be already realigned
+
+    func_imgs: list of string
+        functional imgs path, that will be corrected with topup
+
+    total_readout_times: tuple of floats, optional (default None)
+        the first float corresponds to the total_readout_times of the ap img
+        and the second float to the pa img. Its crucial to provide these if
+        they are different. The times should be provided in seconds.
+
+    memory: Nipype `Memory` object, optional (default None)
+        if given, then caching will be enabled
+
+    tmp_dir: string, optional (default None)
+        temporary directory to store acquisition parameters configuration file
+        for topup. Must be provided in case memory is provided.
+    """
+    if total_readout_times is None:
+        total_readout_times = (1., 1.)  # Check FSL documentation explanation
+
+    if memory is not None:
+        if tmp_dir is None:
+            raise('Temporary directory was not provided with memory object')
+        # Merge AP and PA images
+        merge = memory.cache(fsl.Merge)
+        appa_merged = merge(in_files=[ap_realig_img, pa_realig_img],
+                            dimension='t')
+
```

Comment:
Here, you have to deal with a gotcha of FDL topup: it requires the images to have an even number of slices (sounds crazy, but true indeed). So you probably need to add something like:
```
    odd = (np.mod(nib.load(appa_merged).shape[2], 2) == 1)
    if odd:
        cmd = "fsl5.0-fslroi %s /tmp/pe 0 -1 0 -1 0 1 0 -1" % appa_merged
        commands.getoutput(cmd))
        cmd = "fsl5.0-fslmerge -z %s /tmp/pe %s" % (
            appa_merged, appa_merged)
        print(mem.cache)(commands.getoutput(cmd))
```